File: java/src/main/java/it/giannibombelli/wsc2026/common/domain/primitive/ClientReference.java
package it.giannibombelli.wsc2026.common.domain.primitive;

import it.giannibombelli.wsc2026.common.domain.identity.EntityId;
import it.giannibombelli.wsc2026.common.utils.Require;

import java.util.UUID;

public record ClientReference(UUID value) implements EntityId {
    public ClientReference {
        Require.requireArgument(value, "clientReference");
    }

    @Override
    public String toString() {
        return value.toString();
    }
}

File: typescript/src/common/domain/primitive/clientReference.ts
import { requireArgument } from '@/common/utils/requireArgument.js';
import type { EntityId } from '@/common/domain/identity/entityId.js';
import { Uuid } from '@/common/domain/primitive/uuid.js';

export class ClientReference implements EntityId {
  readonly value: Uuid;

  constructor(value: Uuid) {
    requireArgument(value, 'clientReference');
    this.value = value;
  }

  toString(): string {
    return this.value.value;
  }
}
